mpitest/test_par_doe.py

```diff
@@ -291,33 +303,35 @@ def test_load_balanced_doe_soft_fail(self):
         problem.driver = FullFactorialDriver(num_levels=num_levels,
                                        num_par_doe=self.N_PROCS,
                                        load_balance=True)
+        problem.driver.options['auto_add_response'] = True
+
         problem.driver.add_desvar('indep_var.x',
                                   lower=1.0, upper=float(num_levels))
         problem.driver.add_objective('mult.y')
-
-        problem.driver.add_recorder(InMemoryRecorder())
+        problem.driver.add_response('mult.case_rank')
 
         problem.setup(check=False)
         problem.run()
 
-        num_cases = len(problem.driver.recorders[0].iters)
-
-        if MPI and self.comm.rank > 0:
-            self.assertEqual(num_cases, 0)
-        else:
-            self.assertEqual(num_cases, num_levels)
-
         nfails = 0
+        num_cases = 0
         cases_in_fail_rank = 0
-        for data in problem.driver.recorders[0].iters:
-            if data['success']:
-                self.assertEqual(data['unknowns']['indep_var.x']*2.0,
-                                 data['unknowns']['mult.y'])
+        for responses, success, msg in problem.driver.get_responses():
+            responses = dict(responses)
+            num_cases += 1
+            if success:
+                self.assertEqual(responses['indep_var.x']*2.0,
+                                 responses['mult.y'])
             else:
                 nfails += 1
-            if data['unknowns']['mult.case_rank'] == fail_rank:
+            if responses['mult.case_rank'] == fail_rank:
```

Comment:
this one's actually OK.  I'm counting the total number of cases, both successes and failures, that ran on fail_rank, which is the rank where I triggered failures, but only for a subset of cases that I ran on that rank.